include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -306,6 +306,82 @@ MultigridPreconditionerBase<dim, Number>::check_levels(std::vector<MGLevelInfo>
   }
 }
 
+
+
+template<int dim, int spacedim>
+std::vector<std::shared_ptr<const Triangulation<dim, spacedim>>>
+create_geometric_coarsening_sequence(const Triangulation<dim, spacedim> & fine_triangulation_in)
+{
+  std::vector<std::shared_ptr<const Triangulation<dim, spacedim>>> coarse_grid_triangulations(
+    fine_triangulation_in.n_global_levels());
+
+  coarse_grid_triangulations.back().reset(&fine_triangulation_in, [](auto *) {
+    // empty deleter, since fine_triangulation_in is an external field
+    // and its destructor is called somewhere else
+  });
+
+  // for a single level nothing has to be done
+  if(fine_triangulation_in.n_global_levels() == 1)
+    return coarse_grid_triangulations;
+
+#ifndef DEAL_II_WITH_P4EST
```

Comment:
this negation is difficult to read for me. Why not change the logic?

Reply:
Why did you remove `DEAL_II_WITH_P4EST`?

Reply:
ExaDG requires p4est.

Reply:
Just so that I can follow: We also have p:f:T and metis in ExaDG, so these code branche still require p4est?